tensorflow/core/util/gpu_device_functions.h

```diff
@@ -83,39 +93,52 @@ class CudaGridRange {
   T end_;
 };
 
+#ifndef TENSORFLOW_USE_ROCM
```

Comment:
Do you want CudaGridRange conditionally, or could you just remove the #if/else?

Reply:
I want it conditionally, as in most cases of `CREATE_CUDA_*`. Right now the type is also used in `segment_reduction_ops_gpu.cu.cc` and `trdiagonal_solve_op_gpu.cu.cc` kernels. Having it conditionally available in cuda will make those code compile. 

Sometime in the future when we send those (ROCm available) kernels up to review we will update the symbol to be using GpuGridRange and get rid of the conditional variable here.

Reply:
I would rather be able to change call sites independently of adding ROCm support to the kernels. The two things are unrelated and should not merged into single PRs.

Reply:
Good clarification.

@whchung FYI. Since in our codebase, all call-sites are already renamed. If we start with ROCm support of individual kernels, we will end up doing two things together. Therefore it looks like the right sequence is to:

- File individual PRs to each symbols from `cuda` to `gpu`, one symbol in a PR at a time.
- File individual PRs to enable one kernel as a time.

Reply:
@chsigg, @jerryyin 

I want to make sure we are on the same page here.

Once this PR is merged, we will have the 
1. Gpu* names visiblle in both ROCm and CUDA builds
2. Cuda* names visible only in CUDA builds

Once this PR is merged, we will also start filing PRs for ops that reference these names (in the same 1 PR/op pattern we have established so far). As part of the changes for such a PR, our intent is to update the reference(s) to the Cuda* name(s) to Gpu* name(s).  This implies that the renaming will be spread out across multiple PRs.

If it is preferable to have one PR, that does the Cuda* to Gpu* renaming for all references in a single shot (@chsigg, this is what my interpretation of your comment was...please correct if wrong), then we need to file that PR before we file any other PRs to upstream ROCm support. Remember that Cuda* names are not visible in ROCm builds!

Reply:
> Remember that Cuda* names are not visible in ROCm builds!
Right, I forgot that. So the #ifndef here is consistent with the rest. 

It will be easier if we change the call sites from Cuda* to Gpu* on our side. We have dependencies that you don't see and infrastructure to help making the change. It will be a series of CLs.